From 05a105fea918080ba2fb131a22d8bcca94fc0587 Mon Sep 17 00:00:00 2001
From: Pawel Reich <preich@internships.antmicro.com>
Date: Wed, 20 May 2026 16:27:38 +0200
Subject: [PATCH] Fix `__UINT32_C` not defined

---
 rust/zephyr-sys/wrapper.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/rust/zephyr-sys/wrapper.h b/rust/zephyr-sys/wrapper.h
index 5a855e1..323a92b 100644
--- a/rust/zephyr-sys/wrapper.h
+++ b/rust/zephyr-sys/wrapper.h
@@ -2,6 +2,17 @@
 // defined
 #define _GCC_MAX_ALIGN_T
 
+#if defined(__clang__) && !defined(__UINT32_C)
+#define __INT8_C(c) c
+#define __INT16_C(c) c
+#define __INT32_C(c) c
+#define __INT64_C(c) c##LL
+#define __UINT8_C(c) c
+#define __UINT16_C(c) c
+#define __UINT32_C(c) c##U
+#define __UINT64_C(c) c##ULL
+#endif
+
 #include <version.h>
 #if KERNEL_VERSION_MAJOR < 3
 #include <kernel.h>
-- 
2.54.0

